How to compute min fee (#1732)
LiskHQ/lisk-docs: docs/modules/ROOT/pages/understand-blockchain/blocks-txs.adoc

 15360/153 = 100.39 maximum token transfer transactions per block
====

=== Transaction fees

In order to send and process a transaction, it is required to pay a small fee.

Fees can be defined dynamically for every transaction.

> A higher fee will create an incentive for validators to include the transaction in a block as quickly as possible, because most validators prioritize transactions with higher fees in the transaction pool, because it means a personal higher rewards for them, if they include these transactions in a block.

Reviewer: ```suggestion
A higher fee will create an incentive for validators to include the transaction in a block as quickly as possible, because most validators prioritize transactions with higher fees in the transaction pool, because it means a personal higher reward for them, if they include these transactions in a block.
```

> So by choosing a higher fee, it can be ensured that the transaction is executed as quickly as possible, in case network is crowded.

Reviewer: ```suggestion
So by choosing a higher fee, it can be ensured that the transaction is executed as quickly as possible, in case the network is crowded.
```

This mechanism can also be used to overwrite existing transactions in the pool, by using the same nonce, but a higher fee than the transaction that should be overwritten.

For every transaction, there is a minimum fee that needs to be paid in order for the transaction to be successfully processed.

> The minimum fee required is calculated the following way:

Reviewer: ```suggestion
The minimum fee required is calculated in the following way:
```

 minFee = transactionBytes.length * minFeePerByte

As shown in the formula, the minimum fee is directly connected to the size of the transaction object in bytes after it has been encoded.

Some transaction types may also require an additional fee, called the <<command-fee>>.
The minimum fee for a valid transaction is in this case:

 minFee = transactionBytes.length * minFeePerByte + commandFee

[CAUTION]
====

> * If a transaction requires a command fee, and the command fee is not or only partially paid, the transaction will be <<valid-vs-invalid-transactions,invalid>>.

Reviewer: ```suggestion
* If a transaction requires a command fee, and the command fee is partially paid or not paid at all, then, the transaction will be <<valid-vs-invalid-transactions,invalid>>.
```

* If the transaction is posted with a fee smaller than the required minimum fee, the transaction will <<successful-vs-failed-transactions,fail>>.
====

=== Minimum fee per byte

The minimum fee per byte is defined in the configuration options of the Fee module.

The default value for `minFeePerByte` is 1000 Beddows, or 0.00001 LSK.

> TIP: It is possible to configure a sidechain to have no transactions fees, by setting the `minFeePerByte` to 0 in the config.

Reviewer: ```suggestion
TIP: It is possible to configure a sidechain to have no transaction fee, by setting the `minFeePerByte` to 0 in the config.
```
